libshvbroker/src/brokerapp.cpp

```diff
@@ -663,8 +687,98 @@ class LdapAuthThread : public QThread {
 };
 #endif
 
+#if QT_VERSION_MAJOR >= 6
+namespace azure {
+namespace {
+[[noreturn]] void throw_with_msg(const std::string& msg)
+{
+	throw std::runtime_error{"Azure authorization failed: " + msg};
+}
+}
+}
+
+class AzureAuth : public QObject {
+	Q_OBJECT
+public:
+	AzureAuth(const chainpack::UserLoginContext& ctx, const AzureConfig& azure_config)
+		: m_ctx(ctx)
+		, m_manager(new QNetworkAccessManager(this))
+	{
+		do_request(QUrl{"https://graph.microsoft.com/v1.0/me"}).then([this] (const QJsonDocument& json) {
+			if (!json.object().contains("id")) {
+				azure::throw_with_msg("Couldn't fetch user ID");
+			}
+
+			m_username = json.object()["id"].toString().toStdString();
+			return do_request(QUrl{"https://graph.microsoft.com/v1.0/me/transitiveMemberOf"});
+		}).unwrap().then([this, azure_config] (const QJsonDocument& json) {
```

Comment:
I'm not 100% sure, but my understanding is, that `azure_config` is passed by value here. I also recommend to always use context object, even if it might not be necessary.

Reply:
>context object
Fixed

>azure_config
I have put azure_config into the AzureAuth class and now the lambda just captures `this`. I'm not sure if that's what you meant.